### PR TITLE
NIFI-13356: Protobuf Reader fails on parsing certain repeated fields

### PR DESCRIPTION
**Pull request by mark-bathori**
# Summary

[NIFI-13356](https://issues.apache.org/jira/browse/NIFI-13356)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### UI Contributions

- [ ] NiFi is modernizing its UI. Any contributions that update the [current UI](https://github.com/apache/nifi/tree/main/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui) also need to be implemented in the [new UI](https://github.com/apache/nifi/tree/main/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi).  

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-extension-bundles/nifi-protobuf-bundle/nifi-protobuf-services/src/main/java/org/apache/nifi/services/protobuf/converter/ProtobufDataConverter.java [hunk 1]
```diff
@@ -368,4 +410,30 @@ private RecordSchema generateRecordSchema(String typeName) {
...
     }
+
+    private <T> List<T> processRepeatedValues(CodedInputStream input, ValueReader<CodedInputStream, T> valueReader) {
+        List<T> result = new ArrayList<>();
+        try {
+            while (input.getBytesUntilLimit() > 0) {
+                result.add(valueReader.apply(input));
+            }
+        } catch (Exception e) {
+            throw new IllegalStateException("Unable to parse repeated field");
```
Comment (exceptionfactory) [inline nifi-extension-bundles/nifi-protobuf-bundle/nifi-protobuf-services/src/main/java/org/apache/nifi/services/protobuf/converter/ProtobufDataConverter.java:421]: The root cause should be passed for troubleshooting.
```suggestion
            throw new IllegalStateException("Unable to parse repeated field", e);
```

File: nifi-extension-bundles/nifi-protobuf-bundle/nifi-protobuf-services/src/main/java/org/apache/nifi/services/protobuf/converter/ProtobufDataConverter.java [hunk 2]
```diff
@@ -368,4 +410,30 @@ private RecordSchema generateRecordSchema(String typeName) {
...
+        if (schema.getType(protoType) instanceof MessageType) {
+            return true;
+        }
+
+        if (protoType.isScalar()) {
+            FieldType fieldType = FieldType.findValue(protoType.getSimpleName());
+            return fieldType.equals(STRING) || fieldType.equals(BYTES);
+        } else {
+            return false;
+        }
```
Comment (exceptionfactory) [inline nifi-extension-bundles/nifi-protobuf-bundle/nifi-protobuf-services/src/main/java/org/apache/nifi/services/protobuf/converter/ProtobufDataConverter.java:437]: Recommend refactoring this method slightly to have a single `return` statement. The return value could be set in the appropriate conditional, making it `final` in the initial declaration.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for addressing this issue and implementing additional tests @mark-bathori. The general approach looks good, just noted two minor adjustment recommendations.

**Comment by mark-bathori** [comment]
Thanks @exceptionfactory for the review, I've push a new commit with the requested changes.

File: nifi-extension-bundles/nifi-protobuf-bundle/nifi-protobuf-services/src/main/java/org/apache/nifi/services/protobuf/converter/ProtobufDataConverter.java (hunk 2 above)
Comment (mark-bathori) [reply nifi-extension-bundles/nifi-protobuf-bundle/nifi-protobuf-services/src/main/java/org/apache/nifi/services/protobuf/converter/ProtobufDataConverter.java]: I've refactored the method, let me know what do you think about it.

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustments @mark-bathori, the latest version looks good. +1 merging

**Pull request closed without merging**